amqp/transport.py

```diff
@@ -149,26 +149,43 @@ def having_timeout(self, timeout):
                     sock.settimeout(prev)
 
     def _connect(self, host, port, timeout):
-        entries = socket.getaddrinfo(
-            host, port, 0, socket.SOCK_STREAM, SOL_TCP,
-        )
-        for i, res in enumerate(entries):
-            af, socktype, proto, canonname, sa = res
+        e = None
+
+        # avoid additional DNS requests for AAAA if A succeeds
```

Comment:
If we are going to complicate the logic here, at the very least let's leave a more meaningful comment that walks you through the process.

Reply:
Gotcha. Added some comments in the method.